algorithms/perf_test/test_inclusive_scan.cpp

```diff
@@ -0,0 +1,207 @@
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <cstddef>
+#include <cstdint>
+#include <tuple>
+
+#include <benchmark/benchmark.h>
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Timer.hpp>
+#include <Kokkos_StdAlgorithms.hpp>
+// FIXME: Benchmark_Context.hpp should be moved to a common location
+#include "../../core/perf_test/Benchmark_Context.hpp"
+
+namespace KE = Kokkos::Experimental;
+
+using ExecSpace     = Kokkos::DefaultExecutionSpace;
+using HostExecSpace = Kokkos::DefaultHostExecutionSpace;
+
+template <class ValueType>
+struct SumFunctor {
+  KOKKOS_FUNCTION
+  ValueType operator()(const ValueType& a, const ValueType& b) const {
+    return (a + b);
+  }
+};
+
+template <class ValueType>
+struct MaxFunctor {
+  KOKKOS_FUNCTION
+  ValueType operator()(const ValueType& a, const ValueType& b) const {
+    if (a > b)
+      return a;
+    else
+      return b;
+  }
+};
+
+// Helper to obtain last element of a view
+template <class T>
+T obtainLastElem(const Kokkos::View<T*, ExecSpace>& v) {
+  T last_element;
+  Kokkos::deep_copy(last_element, Kokkos::subview(v, v.extent(0) - 1));
+  return last_element;
+}
+
+// Helper to allocate input and output views
+template <class T>
+auto prepareViews(const std::size_t kProbSize) {
+  Kokkos::View<T*, ExecSpace> in{"input", kProbSize};
+  Kokkos::View<T*, ExecSpace> out{"output", kProbSize};
+
+  auto h_in = Kokkos::create_mirror_view(in);
+
+  for (std::size_t i = 0; i < kProbSize; ++i) {
+    h_in(i) = i;
+  }
+
+  Kokkos::deep_copy(in, h_in);
+
+  return std::make_tuple(in, out, h_in);
+}
+
+// Perform scan in a serial manner. Used as reference.
+template <class T, class ScanFunctor = SumFunctor<T>>
+T serialScan(const Kokkos::View<T*, HostExecSpace>& h_in,
+             ScanFunctor scan_functor = ScanFunctor()) {
+  std::size_t view_size = h_in.extent(0);
+
+  Kokkos::View<T*, HostExecSpace> h_out("output", view_size);
+
+  h_out(0) = h_in(0);
+
+  for (std::size_t i = 1; i < view_size; ++i) {
+    h_out(i) = scan_functor(h_in(i), h_out(i - 1));
+  }
+
+  return h_out(view_size - 1);
+}
+
+// Default prefix sum scan
+template <class T>
+auto InclusiveScan(const Kokkos::View<T*, ExecSpace>& in,
+                   const Kokkos::View<T*, ExecSpace>& out, T res_check) {
+  ExecSpace().fence();
+  Kokkos::Timer timer;
+  KE::inclusive_scan("Default scan", ExecSpace(), KE::cbegin(in), KE::cend(in),
+                     KE::begin(out));
+  ExecSpace().fence();
+  double time_scan = timer.seconds();
+
+  T res_scan  = obtainLastElem(out);
+  bool passed = (res_scan == res_check);
+
+  return std::make_tuple(time_scan, passed);
+}
```

Comment:
could you unify this with the non-default below? At the moment this is repeating a lot of code